api/python/tests/integration/test_packages.py

```diff
@@ -1410,3 +1424,68 @@ def test_resolve_hash(self):
 
         with pytest.raises(QuiltException, match='Found multiple matches'):
             Package.resolve_hash(pkg_name, LOCAL_REGISTRY, hash_prefix)
+
+
+# The following tests were moved out of the PackageTest class to enable parametrization.
+# see (https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses)
+@pytest.mark.parametrize(
+    'update_policy, expected_one_byte',
+    [
+        ('incoming', b'one'),
+        ('existing', b'1')
+    ]
+)
+def test_set_dir_update_policy(update_policy, expected_one_byte):
+    """Verify building a package with update policy. """
+    data_dir = pathlib.Path(__file__).parent / "data"
+    nested_dir = data_dir / 'nested'
+    pkg = Package()
+    pkg.set_dir("/", nested_dir, meta={'name': 'test_meta'})
+    assert 'one.txt' in pkg.keys()
+    assert 'two.txt' not in pkg.keys()
+    assert 'three.txt' not in pkg.keys()
+    assert pkg.meta == {'name': 'test_meta'}
+
+    nested_dir_2 = data_dir / 'nested2'
+    pkg.set_dir("/", nested_dir_2, update_policy=update_policy)
+    assert pkg['one.txt'].get_bytes() == expected_one_byte
+    assert pkg['two.txt'].get_bytes() == b'2'
+    assert pkg['three.txt'].get_bytes() == b'3'
+
+
+@pytest.mark.parametrize(
+    'update_policy, expected_a_url, expected_xy_url',
+    [
+        ('existing', 's3://bucket/foo/a.txt?versionId=xyz', 's3://bucket/foo/x/y.txt?versionId=null'),
+        ('incoming', 's3://bucket/bar/a.txt?versionId=abc', 's3://bucket/bar/x/y.txt?versionId=null')
+    ]
+)
+def test_set_dir_update_policy_s3(update_policy, expected_a_url, expected_xy_url):
+    with patch('quilt3.packages.list_object_versions') as list_object_versions_mock:
+        list_object_versions_mock.return_value = (
+            [
+                dict(Key='foo/a.txt', VersionId='xyz', IsLatest=True, Size=10),
```

Comment:
to be more realistic we really don't want to have the exact same files and folders in `foo` and `bar`. you can think of test coverage in this way: test when SOME keys overlap, when no keys overlap, when all keys overlap.